rs/bitcoin/kyt/src/providers/tests.rs

```diff
@@ -0,0 +1,29 @@
+use super::*;
+
+#[test]
+fn test_parse_authorization_header_from_url() {
+    let result = parse_authorization_header_from_url("http://localhost:3030");
+    assert!(matches!(result, Err(err) if err.contains("Missing username")));
```

Comment:
nit (here and below): using [`assert_matches` ](https://docs.rs/assert_matches/1.5.0/assert_matches/) would improve debuggability since `assert!` just says something like `true != false`

Reply:
Good to know! Thanks

Reply:
Thanks for having looked into it!